File: frontend/src/api/feedback.ts
import apiClient from './client';
import { Feedback } from '../types';

export interface SubmitFeedbackRequest {
  type: 'bug' | 'feature' | 'improvement';
  title: string;
  description: string;
  email?: string;
}

export const feedbackApi = {
  // 提交反馈
  submitFeedback: async (data: SubmitFeedbackRequest) => {
    const response = await apiClient.post<Feedback>('/api/feedback', data);
    return response.data;
  },
};

File: backend/app/schemas/feedback.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class FeedbackCreate(BaseModel):
    type: str  # bug, feature, improvement
    title: str
    description: str
    email: Optional[str] = None

class Feedback(FeedbackCreate):
    id: str
    created_at: datetime

    class Config:
        from_attributes = True